support/k8s/k8s-workload-registrar/config_crd.go

```diff
@@ -90,11 +101,71 @@ func (c *CRDMode) Run(ctx context.Context) error {
 	}
 
 	if c.WebhookEnabled {
+		// Backwards compatibility check
+		exists, err := c.certDirExistsAndReadOnly()
+		if err != nil {
+			return fmt.Errorf("checking webhook certificate directory permissions: %w", err)
+		}
+		if exists {
+			log.Warn("Detected statically mounted webhook certificate directory, support for this will be removed in a future version. " +
+				"Refer to README for instructions on using SPIRE Server to populate webhook certificates.")
+		} else {
+			myNodeName, err := getMyNodeName()
+			if err != nil {
+				return err
+			}
+			myPodName, err := getMyPodName()
+			if err != nil {
+				return err
+			}
+			myPodUID, err := getMyPodUID()
+			if err != nil {
+				return err
+			}
+			clientset, err := controllers.NewKubeClientset()
+			if err != nil {
+				return err
+			}
+			webhookEntry := webhook.NewEntry(webhook.EntryConfig{
+				Clientset:          clientset,
+				Cluster:            c.Cluster,
+				Ctx:                ctx,
+				E:                  entryClient,
+				Log:                log,
+				Name:               myPodName,
+				Namespace:          myPodNamespace,
+				NodeName:           myNodeName,
+				TrustDomain:        c.TrustDomain,
+				UID:                myPodUID,
+				WebhookServiceName: c.WebhookServiceName,
+			})
+			if err = webhookEntry.CleanupStaleEntries(); err != nil {
+				return err
+			}
+			if err = webhookEntry.CreateEntry(); err != nil {
+				return err
+			}
+			defer func() {
+				if err = webhookEntry.DeleteEntry(); err != nil {
+					log.WithError(err).Error("Unable to delete webhook entry")
+				}
+			}()
+			webhookSVIDWatcher := webhook.NewSVIDWatcher(webhook.SVIDWatcherConfig{
+				AgentSocketPath: c.AgentSocketPath,
```

Comment:
The agent socket path is not a required configurable. It is currently only used when the registrar is not colocated with SPIRE server and needs to obtain an admin SVID. Up until now there has been no prerequisite that the registrar be colocated with an agent. From what I understand, this change requires that the registrar go through an agent to obtain an SVID for hosting the webhook? I wonder if the mint API would be appropriate to use instead. Paging @evan2645.

Reply:
Yes, we use the agent to get the SVID for the webhook. Is there a way to keep the SVIDs up to date with the mint API as the agent does?

Reply:
Not without manually tracking the expiration and requesting a new SVID before the old one expires. Since this would involve a single identity, it doesn't seem too onerous to do that. Curious what @evan2645 thinks.